cmd/mesh/profile-dump.go

```diff
@@ -88,6 +89,21 @@ func profileDump(args []string, rootArgs *rootArgs, pdArgs *profileDumpArgs, l *
 
 func genProfile(helmValues bool, inFilename, profile, setOverlayYAML, configPath string) (string, error) {
 	overlayYAML := ""
+	profiles := helm.ListBuiltinProfiles()
```

Comment:
This is not quite right as it only handles "profile" field. 
If you look at manifestApply, there's this line: 
	overlayFromSet, err := makeTreeFromSetList(maArgs.set)
This is where --set processing is done. If it's not working, something has been broken here.

Reply:
makeTreeFromSetList has no issue, it has successfully parsed the `profile` arg and put it into overlayFromSet.

The issue here is we always use `""` as the `profile` value and ignore the values from overlayFromset.  All other values from set will be respected but only `profile` will be ignored.

Reply:
Ok, I understand the problem now. I think it would be better to work with YAML rather than text below e.g. 
- unmarshal setOverlayYAML
- if profile is unset, set it to yamlTree["profile"]

Reply:
Thanks Martin and Fixed